lib/diagnostic.rb

```diff
@@ -7,36 +7,39 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    list.max_by(&:length)
+    # my second try would be list.inject { |f, s| f.length > s.length ? f : s }
   end
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    list.reduce(:+)
   end
 
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
   def some_odd(numbers)
-    # Your code goes here
+    numbers.any?(&:odd?)
   end
 
   # Question 4
   # Write ruby code that returns true if every entry in the
   # list of numbers are even
   def every_even(numbers)
-    # Your code goes here
+    if numbers.some_odd == false
+       p true
+    end
   end
```

Comment:
2 things: 1) what is `.some_odd`? does it exist? 2) look at what the question says: "returns true if..." What is your method doing? It's printing out true.